ffmpeg_plugin/0001-lavc-svt_hevc-add-libsvt-hevc-encoder-wrapper.patch

```diff
@@ -377,38 +377,23 @@ index 0000000000..aa51830788
 +    }
 +
 +    if (avctx->flags & AV_CODEC_FLAG_GLOBAL_HEADER) {
-+        EB_BUFFERHEADERTYPE headerPtr;
-+        headerPtr.nSize       = sizeof(headerPtr);
-+        headerPtr.nFilledLen  = 0; /* in/out */
-+        headerPtr.pBuffer     = av_malloc(10 * 1024 * 1024);
-+        headerPtr.nAllocLen   = (10 * 1024 * 1024);
-+
-+        if (!headerPtr.pBuffer) {
-+            av_log(avctx, AV_LOG_ERROR,
-+                   "Cannot allocate buffer size %d.\n", headerPtr.nAllocLen);
-+            svt_ret = EB_ErrorInsufficientResources;
-+            goto failed_init_enc;
-+        }
++        EB_BUFFERHEADERTYPE *headerPtr = NULL;
 +
 +        svt_ret = EbH265EncStreamHeader(svt_enc->svt_handle, &headerPtr);
 +        if (svt_ret != EB_ErrorNone) {
 +            av_log(avctx, AV_LOG_ERROR, "Error when build stream header.\n");
-+            av_freep(&headerPtr.pBuffer);
 +            goto failed_init_enc;
 +        }
 +
-+        avctx->extradata_size = headerPtr.nFilledLen;
++        avctx->extradata_size = headerPtr->nFilledLen;
 +        avctx->extradata = av_mallocz(avctx->extradata_size + AV_INPUT_BUFFER_PADDING_SIZE);
 +        if (!avctx->extradata) {
 +            av_log(avctx, AV_LOG_ERROR,
 +                   "Cannot allocate HEVC header of size %d.\n", avctx->extradata_size);
-+            av_freep(&headerPtr.pBuffer);
 +            svt_ret = EB_ErrorInsufficientResources;
 +            goto failed_init_enc;
 +        }
-+        memcpy(avctx->extradata, headerPtr.pBuffer, avctx->extradata_size);
-+
-+        av_freep(&headerPtr.pBuffer);
++        memcpy(avctx->extradata, headerPtr->pBuffer, avctx->extradata_size);
```

Comment:
Does pBuffer need to be freed as before?

Reply:
We don't link the buffer to the package data anymore. Hence, don't need it.